clients/src/main/java/org/apache/kafka/common/telemetry/internals/TelemetryMetricNamingConvention.java

```diff
@@ -111,14 +114,16 @@ private static String cleanMetric(String metric) {
      * @return the new map with keys replaced by snake_case representations.
      */
     private static Map<String, String> cleanTags(Map<String, String> raw) {
-        return raw.entrySet()
-            .stream()
-            .collect(Collectors.toMap(s -> clean(s.getKey(), TAG_JOINER), Entry::getValue));
+        Map<String, String> result = new HashMap<>();
+        raw.forEach((k, v) -> {
+            result.put(clean(k, TAG_JOINER).intern(), v.intern());
+        });
+        return result;
     }
 
     private static String clean(String raw, String joiner) {
         Objects.requireNonNull(raw, "metric data cannot be null");
         String lowerCase = raw.toLowerCase(Locale.ROOT);
-        return lowerCase.replaceAll("-", joiner);
+        return DASH_PATTERN.matcher(lowerCase).replaceAll(joiner);
```

Comment:
Out of curiosity, what prompted this switch from string method to regex?

Reply:
Unfortunately `String.replaceAll` *is* a regex method, without a compiled pattern. Not great API design but it is what it is. So this should be equivalent, but with a pre-compiled pattern, like is done in other places.